hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -257,7 +257,8 @@ public static Map<String, HoodieColumnRangeMetadata<Comparable>> collectColumnRa
       List<Pair<String, Schema.Field>> targetFields,
       String filePath,
       Schema recordSchema,
-      StorageConfiguration<?> storageConfig) {
+      StorageConfiguration<?> storageConfig,
+      HoodieIndexVersion indexVersion) {
```

Comment:
Could the col stats V1 and V2 logic be separated into two methods, `collectColumnRangeMetadataV1` and `collectColumnRangeMetadataV2`?  In the future after #13226 is landed, the logic should be contained in `ColumnStatsIndexer` and there will be version-specific indexer implementation.  So by doing this, it's easier for future refactoring.